--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -45,6 +_,8 @@
 import java.util.logging.Logger;
 import java.util.stream.Collectors;
 import javax.imageio.ImageIO;
+
+import io.rolia.secureseed.Globals;
 import net.md_5.bungee.api.chat.BaseComponent;
 import net.minecraft.Optionull;
 import net.minecraft.advancements.AdvancementHolder;
@@ -1265,7 +_,7 @@
             .result()
             .orElse(null);
         if (worldGenSettings == null) {
-            WorldOptions worldOptions = new WorldOptions(creator.seed(), creator.generateStructures(), creator.bonusChest());
+            WorldOptions worldOptions = new WorldOptions(creator.seed(), Globals.worldSeed, creator.generateStructures(), creator.bonusChest()); // Rolia - keep the WorldCreator bonus-chest flag
 
             DedicatedServerProperties.WorldDimensionData properties = new DedicatedServerProperties.WorldDimensionData(GsonHelper.parse((creator.generatorSettings().isEmpty()) ? "{}" : creator.generatorSettings()), creator.type().name().toLowerCase(Locale.ROOT));
             WorldDimensions worldDimensions = properties.create(context.datapackWorldgen());
